Fabric.Identity.API/scripts/tests/CatalystDosIdentity.tests.ps1

```diff
@@ -451,12 +451,289 @@ Describe 'Test-IsClientRegistered Unit Tests' -tag 'Unit' {
     }
 }
 
-Describe 'Get-ApiRegistration' {}
+Describe 'Get-ApiRegistration Unit Tests' -tag "Unit" {
+    Context 'Valid Request' {
+        It 'Should Return an api object when valid response' {
+            [Uri] $mockUrl = "http://some.server/identity"
+            Mock -ModuleName CatalystDosIdentity -CommandName Invoke-RestMethod { return @{
+                enabled = 1
+                name = "test-Api"
+                scopes = @{"name" = "test-Api"; "displayName" = "Test-API"}
+                userClaims = @("name", "email", "role", "groups")
+            }}
+
+            $response = Get-ApiRegistration -identityUrl $mockUrl -apiName "someApi" -accessToken  "Bearer goodtoken"
+
+            $response.name | Should -Be "test-Api"
+            $response.userClaims.Length | Should -Be 4
+            $response.scopes.Count | Should -Be 2
+        }}
+    Context 'Invalid Request' {
+        It 'Should return an error when non existent api' {
+            [Uri] $mockUrl = "http://some.server/identity"
+            Mock -ModuleName CatalystDosIdentity -CommandName Invoke-RestMethod { throw  New-Object -TypeName "System.Net.WebException" -ArgumentList "Exception" }
+
+            {Get-ApiRegistration -identityUrl $mockUrl -apiName "someNonExistentApi" -accessToken  "Bearer goodtoken"} | Should -Throw "Exception" -ExceptionType System.Net.WebException
+        }
+    }
+}
+
+Describe 'New-ApiRegistration Unit Tests' -tag "Unit" {
+    Context 'Valid Request' {
+        It 'Should Create and Return an api object when valid response' {
```

Comment:
looks like it's returning a secret and not an object

Reply:
the secret is what is returned, not the whole object.

Reply:
yah. I will remove the word object and change to secret